helix-term/src/handlers/completion/resolve.rs

```diff
@@ -26,6 +27,15 @@ pub struct ResolveHandler {
     resolver: Sender<ResolveRequest>,
 }
 
+macro_rules! lsp_variant {
```

Comment:
I'd like to avoid the macro here if possible - I think we can be explicit in the places where we want to match the `CompletionItem::Lsp` variant instead. For `ensure_item_resolved` that means we need to move the `if self.last_request.as_deref()...` check to the top of the function because of the mutable borrow of `item` and then we can say

```rust
let CompletionItem::Lsp(lsp_item) = item else {
    return;
}
```

and we can unconditionally pass the `item` to `ensure_item_resolved` in `helix-term/src/ui/completion.rs`. In the `ResolveRequest::execute` function we can do basically the same but use `unreachable!` (the same as what you have now but inlining the macro)

Reply:
Before I'm repeating work again, do you mean something like [this](https://github.com/helix-editor/helix/pull/2608/commits/cc19db1e5306e6cbb396ae33c51e99927e879584#diff-3626b17eb15f4783dd83a9f709ecef3bd3f591db4b22d83ee2603ff0a09d49e5) before that micro-optimization (I wasn't sure about it anyway, as it just saves 1 clone...)

Reply:
Yeah an `impl PartialEq<CompletionItem> for LspCompletionItem` is much better than that macro, for whatever reason, I haven't thought of that first...

I've reverted to that previous change, and added `PartialEq` impls.